src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -471,13 +471,15 @@ private void closeMappedEntity(final MappedEntity mappedEntity) {
             final PreparedStatement insertReturning = mappedEntity.getInsertReturning();
             final PreparedStatement insertWithAutoInc = mappedEntity.getInsertWithAutoInc();
 
-            insert.executeBatch();
+            if (!insert.isClosed()) {
```

Comment:
Don't we need to do like the others?
`if (insert != null && !insert.isClosed())`